src/pycram/pose_generator_and_validator.py

```diff
@@ -102,137 +103,103 @@ def generate_orientation(position: List[float], origin: Pose) -> List[float]:
         return quaternion
 
 
-def visibility_validator(pose: Pose,
-                         robot: Object,
-                         object_or_pose: Union[Object, Pose],
-                         world: World) -> bool:
+def visibility_validator(robot: Object,
+                         pose: Pose) -> bool:
     """
     This method validates if the robot can see the target position from a given
     pose candidate. The target position can either be a position, in world coordinate
     system, or an object in the World. The validation is done by shooting a
     ray from the camera to the target position and checking that it does not collide
     with anything else.
 
-    :param pose: The pose candidate that should be validated
     :param robot: The robot object for which this should be validated
     :param object_or_pose: The target position or object for which the pose candidate should be validated.
-    :param world: The World instance in which this should be validated.
     :return: True if the target is visible for the robot, None in any other case.
     """
+    world = robot.world
     robot_pose = robot.get_pose()
-    if isinstance(object_or_pose, Object):
-        robot.set_pose(pose)
-        camera_pose = robot.get_link_pose(RobotDescription.current_robot_description.get_camera_link())
-        robot.set_pose(Pose([100, 100, 0], [0, 0, 0, 1]))
-        ray = world.ray_test(camera_pose.position_as_list(), object_or_pose.get_position_as_list())
-        res = ray.obj_id == object_or_pose.id
-    else:
-        robot.set_pose(pose)
-        camera_pose = robot.get_link_pose(RobotDescription.current_robot_description.get_camera_link())
-        robot.set_pose(Pose([100, 100, 0], [0, 0, 0, 1]))
-        # TODO: Check if this is correct
-        ray = world.ray_test(camera_pose.position_as_list(), object_or_pose)
-        res = not ray.intersected
+
+    camera_pose = robot.get_link_pose(RobotDescription.current_robot_description.get_camera_link())
+    robot.set_pose(Pose([100, 100, 0], [0, 0, 0, 1]))
+    ray = world.ray_test(camera_pose.position_as_list(), pose)
+
     robot.set_pose(robot_pose)
-    return res
+    return ray.intersected
```

Comment:
This would return True if the ray intersected any object which is exactly the opposite. It should be "return not ray.intersected"

Reply:
A ray can't intersect with a pose only with objects, this should maybe use something like the validate in LookAt where you spawn a generic object in the prospection world and use that for the rayTest